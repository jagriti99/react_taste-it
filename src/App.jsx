import './App.css';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Layout from './pages/Layout.jsx';
import Home from './components/Home';
import Recipes from './components/Recipes';
import AddNewRecipe from './components/AddNewRecipe';
import SeemoreRecipes from './components/SeemoreRecipes';

const App = () => {
  return (
    <BrowserRouter>       
      <Routes>
        <Route path='/' element={<Layout />}>
          <Route index element={<Home />}></Route>
          <Route path='Recipes' element={<Recipes />} />
          <Route path='/AddNewRecipe' element={<AddNewRecipe />}/>
          <Route path='/recipes/:id' element={<SeemoreRecipes />}/>
        </Route>
      </Routes>         
    </BrowserRouter>
  );
}

export default App;
