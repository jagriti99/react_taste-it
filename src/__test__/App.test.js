import renderer from "react-test-renderer";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import App from "../App";

const recipeDetails = {
  name: "Idly sambhar",
  author: "Jagriti",
  countryFlagurl: "https://flagcdn.com/in.svg",
  description: "tasty",
  image:
    "https://foodhistoria.com/wp-content/uploads/2020/09/Ghee-Sambar-Idli-5-e1600157946772.jpg",
  instructions: "make at home",
  id: 1,
  ingredients: [
    {
      quantity: "3 cup",
      ingredient: "Toor dal",
    },
    {
      quantity: "3 cup",
      ingredient: "Rawa",
    },
    {
      quantity: "1 cup",
      ingredient: "Water",
    },
    {
      quantity: "1.5 cup",
      ingredient: "Curd",
    },
  ],
};

const server = setupServer(
  rest.get("http://localhost:3001/recipes", (req, res, ctx) => {
    return res(ctx.json([recipeDetails]));
  }),
  rest.get("http://localhost:3001/recipes/1", (req, res, ctx) => {
    return res(ctx.json(recipeDetails));
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe("App", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("loads and recipes are browsable", async () => {
    // ARRANGE
    render(<App />);

    await userEvent.click(screen.getByText("Browse Recipes"));
    await waitFor(() => {
      expect(screen.getByText("Idly sambhar")).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.getByText("See more")).toBeInTheDocument();
    });
  });
});
