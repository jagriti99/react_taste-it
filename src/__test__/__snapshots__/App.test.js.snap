// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <header>
    <nav
      className="nav"
    >
      <div
        className="logo"
      >
        <a
          aria-current="page"
          className="taste active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt=""
          />
          Taste-it
        </a>
      </div>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className=""
            href="/Recipes"
            onClick={[Function]}
          >
            Recipes
          </a>
        </li>
        <li>
          <a
            className=""
            href="/AddNewRecipe"
            onClick={[Function]}
          >
            Add New Recipe
          </a>
           
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <div
      className="video_container"
    >
      <video
        autoPlay={true}
        className="video"
        loop={true}
        muted={true}
      >
        <source
          src="food.mp4"
          type="video/mp4"
        />
      </video>
      <div
        className="heading"
      >
        <h2>
          Want to try something new?
        </h2>
        <h1>
          Taste-It
        </h1>
        <p>
          Explore and add recipes
        </p>
        <button
          className="btn"
        >
          <a
            className=""
            href="/Recipes"
            onClick={[Function]}
          >
            Browse Recipes
          </a>
        </button>
      </div>
      <div
        className="container"
      >
        <div
          className="moreDetails"
        >
          <h3>
            Browse recipe
          </h3>
          <p>
            "Search recipes by name or country."
          </p>
          <a
            className=""
            href="/recipes"
            onClick={[Function]}
          >
            All recipes
          </a>
        </div>
        <div
          className="moreDetails"
        >
          <h3>
            Add recipes
          </h3>
          <p>
            "Want to add your own recipe? No worries, add on!"
          </p>
          <a
            className=""
            href="/addNewRecipe"
            onClick={[Function]}
          >
            All recipes
          </a>
        </div>
        <div
          className="moreDetails"
        >
          <h3>
            Browse recipe
          </h3>
          <p>
            "Search recipes by name or country."
          </p>
          <a
            href="https://www.bc.fi"
            rel="noreferrer"
            target="_blank"
          >
            Business College Helsinki hompage
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
